lib/LoginStrategy.ts

```diff
@@ -239,4 +252,34 @@ export class LoginStrategy extends BaseStrategy {
       return acc
     }, [] as Claim[])
   }
-}
+
+  /**
+   * @description checks if the claimer is authenticated into the DID Document
+   *
+   * @param token
+   * @param payload
+   *
+   * @returns {string} issuer DID or empty string
+   */
+  async verifyDidClaim(token: string, { iss }: ITokenPayload): Promise<string> {
+    
+    const didDocument = await this.didResolver.read(iss)
+
+    if (this.isAuthorized(token, didDocument))
+      return iss
+    return null 
+  }
+
+  private isAuthorized = (claimedToken: string,didDocument: IDIDDocument) : boolean => {
+
+    //retrieve claim infos from token
+    const claimInfos: any = decodeJwToken(claimedToken)
+
+    //extract all authenticated users and filter to keep those owning the claimed DID
+    const authenticated = didDocument.authentication.filter(auth => {
+      auth["publicKey"] === `${claimInfos.did}#owner`.toString()
+    })
+
+    return authenticated.length === 0
+  }
+}
```

Comment:
A solid first attempt 👍  but not quite correct

So our DID documents, entries in `authentication` property provide links to entries in the `publicKey` property. So our current DID documents are like this:
```
{
  "id": "did:ethr:0x96142FAC151454E38B897cE50fdE26C60622B14E",
  "service": [],
  "authentication": [
    {
      "type": "owner",
      "validity": {
        "_hex": "0x1fffffffffffff"
      },
      "publicKey": "did:ethr:0x96142FAC151454E38B897cE50fdE26C60622B14E#owner"
    }
  ],
  "publicKey": [
    {
      "id": "did:ethr:0x96142FAC151454E38B897cE50fdE26C60622B14E#key-owner",
      "type": "Secp256k1veriKey",
      "controller": "0x96142FAC151454E38B897cE50fdE26C60622B14E",
      "publicKeyHex": "0x035f58d27f27944f42089083a6c96b099360bb6099988bcfe3ddca74541c0a097d"
    }
  ]
}
```
So in this example, `"publicKey": "did:ethr:0x96142FAC151454E38B897cE50fdE26C60622B14E#owner"` in the `authentication` section links to the key with `"id": "did:ethr:0x96142FAC151454E38B897cE50fdE26C60622B14E#key-owner"` in the `publicKey` property (both should end with `#key-owner`, this is a bug in ew-did-registry that you should account for. The actual public key value that should be used to verify the signature though is `"publicKeyHex": "0x035f58d27f27944f42089083a6c96b099360bb6099988bcfe3ddca74541c0a097d"`.

In other words, the `publicKey` `id` is arbitrary. The code should be able to verify a signature using a DID document like this:
```
{
  "id": "did:ethr:0x96142FAC151454E38B897cE50fdE26C60622B14E",
  "service": [],
  "authentication": [
    {
      "type": "owner",
      "validity": {
        "_hex": "0x1fffffffffffff"
      },
      "publicKey": "my-favorite-public-key-1"
    }
  ],
  "publicKey": [
    {
      "id": "my-favorite-public-key-1",
      "type": "Secp256k1veriKey",
      "controller": "0x96142FAC151454E38B897cE50fdE26C60622B14E",
      "publicKeyHex": "0x035f58d27f27944f42089083a6c96b099360bb6099988bcfe3ddca74541c0a097d"
    }
  ]
}
```

Reply:
That Makes a lot of sens, and gets clearer in my mind! Thank you for your careful review and your precious explanations John. I'm on it !